src/views/Generator/TestOptions/DataFiles.tsx

```diff
@@ -0,0 +1,84 @@
+import { FieldGroup } from '@/components/Form'
+import { TestDataSchema } from '@/schemas/generator'
+import { useGeneratorStore } from '@/store/generator'
+import { useStudioUIStore } from '@/store/ui'
+import { TestData } from '@/types/testData'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { CheckboxGroup, Text } from '@radix-ui/themes'
+import { useCallback, useEffect } from 'react'
+import { useForm } from 'react-hook-form'
+
+export function DataFiles() {
+  const availableFiles = useStudioUIStore((store) => store.dataFiles)
+  const files = useGeneratorStore((store) => store.files)
+  const setFiles = useGeneratorStore((store) => store.setFiles)
+
+  const {
+    register,
+    handleSubmit,
+    watch,
+    setValue,
+    formState: { errors },
+  } = useForm<Pick<TestData, 'files'>>({
+    resolver: zodResolver(TestDataSchema.pick({ files: true })),
+    shouldFocusError: false,
+    defaultValues: {
+      files,
+    },
+  })
+
+  const options = [...availableFiles.values()].map((file) => ({
+    label: file.displayName,
+    value: file.displayName,
+  }))
+  const value = watch('files').map(({ name }) => name)
+
+  const onSubmit = useCallback(
+    (data: Pick<TestData, 'files'>) => {
+      setFiles(data.files)
+    },
+    [setFiles]
+  )
+
+  // Submit onChange
+  useEffect(() => {
+    const subscription = watch(() => handleSubmit(onSubmit)())
+    return () => subscription.unsubscribe()
+  }, [watch, handleSubmit, onSubmit])
+
+  const handleChange = (value: string[]) => {
+    setValue(
+      'files',
+      value.map((fileName) => ({
+        name: fileName,
+      }))
+    )
+  }
+
+  const filesField = register('files')
```

Comment:
I probably don't know how to cook `@hookform`, but this feels excessive for what I'm trying to achieve here 😞

Reply:
A controlled `CheckboxGroup` could also work here and we may be able to remove the `watch` and `register` calls.

```tsx
<Controller
   name="files"
   control={control}
   render={({ field }) => (
       <CheckboxGroup.Root
         value={field.value}
         onValueChange={field.onChange}
        >
           {options.map((option) => .... }
      </CheckboxGroup.Root>
   )}
/>
```

Reply:
So, I tried something like this, but the problem I ran into was that that field has type `TestData`, but `CheckboxGroup` value is `string[]`. Maybe I can resolve that on submit?

Reply:
Ah I see, yes in that case we may have to transform them `onSubmit` before saving it to file.